src/CMakeLists.txt

```diff
@@ -0,0 +1,179 @@
+	quick_divcon_module.f90 quick_ecp_module.f90 quick_electrondensity_module.f90 
+	quick_files_module.f90 quick_timer_module.f90 quick_gridpoints_module.f90 
+	quick_mfcc_module.f90 quick_params_module.f90 quick_pb_module.f90 
+	quick_scratch_module.f90 quick_all_module.f90 quick_scf_module.f90 
+	quick_gradient_module.f90 quick_api_module.f90 quick_api_test_module.f90)
+
+set(QUICK_SUBS_SOURCES Angles.f90 copyDMat.f90 copySym.f90 
+	degen.f90 denspt.f90 diag.f90 dipole.f90 
+	EffChar.f90 eigvec.f90 
+	ekinetic.f90 findBlock.f90 fmt.f90 getinum.f90 
+	getNum.f90 greedy_distrubute.f90 hrr.f90 iatoi.f90 
+	iatoimp.f90 io.f90 iwhole.f90 
+	lbfgs.f90 Lsolve.f90 matComp.f90 matMul.f90 
+	order.f90 orthog.f90 PriCol.f90 PriSym.f90 
+	PrtAct.f90 PrtDat.f90 PrtErr.f90 PrtLab.f90 
+	PrtMsg.f90 PrtTim.f90 PrtWrn.f90 pteval.f90 
+	quick_open.f90 random.f90 rdinum.f90 rdnml.f90 
+	rdnum.f90 rdword.f90 readPDB.f90 spdfgh.f90 
+	ssw.f90 sum2Mat.f90 transpose.f90 tridi.f90 
+	upcase.f90 vett.f90 whatis.f90 whole.f90 
+	wrtRestart.f90 xnorm.f90 zeroMatrix.f90 zmake.f90 
+	pt2der.f90 sswder.f90 denspt_new_imp.f90 
+	pteval_new_imp.f90 scaMatMul.f90)
+
+set(QUICK_GENERAL_SOURCES initialize.f90 read_job_and_atom.f90 fmm.f90 
+    getMolSad.f90 getMol.f90 shell.f90 schwarz.f90 
+    quick_one_electron_integral.f90 getEnergy.f90 inidivcon.f90 
+    ecp.f90 hfoperator.f90 nuclear.f90 
+    dft.f90 sedftoperator.f90 
+    scf.f90 uscf.f90 finalize.f90 uhfoperator.f90 
+    udftoperator.f90 usedftoperator.f90 
+    uelectdii.f90 mpi_setup.f90 quick_debug.f90 
+    calMP2.f90 optimize.f90 gradient.f90 hessian.f90 
+    CPHF.f90 frequency.f90 MFCC.f90 basis.f90 
+    fake_amber_interface.f90 scf_operator.f90)
+
+set(TEST_API_SOURCES quick_api_test.f90)
+set(MAIN_SOURCES main.f90)
+
+
+# the "subs" and "modules" source lists are for items in subdirectories.  We have to prepend the subdirectory path to these items.
+# --------------------------------------------------------------------
+
+set(QUICK_MODULES_SOURCES_FULLPATH "")
+set(QUICK_SUBS_SOURCES_FULLPATH "")
+
+foreach(MODULES_SOURCE ${QUICK_MODULES_SOURCES})
+	list(APPEND QUICK_MODULES_SOURCES_FULLPATH "modules/${MODULES_SOURCE}")
+endforeach()
+
+foreach(SUBS_SOURCE ${QUICK_SUBS_SOURCES})
+	list(APPEND QUICK_SUBS_SOURCES_FULLPATH "subs/${SUBS_SOURCE}")
+endforeach()
+
+# create serial and MPI libraries
+# --------------------------------------------------------------------
+
+add_library(libquick ${QUICK_MODULES_SOURCES_FULLPATH} ${QUICK_SUBS_SOURCES_FULLPATH} ${QUICK_GENERAL_SOURCES})
+target_compile_options(libquick PRIVATE ${OPT_FFLAGS})
+config_module_dirs(libquick ${QUICK_SERIAL_MOD_DIR} ${XC_SERIAL_MOD_DIR})
+target_link_libraries(libquick xc octree blas)
+install_libraries(libquick)
+
+# "liblibquick" -> "libquick"
+set_property(TARGET libquick PROPERTY OUTPUT_NAME quick)
+
+if(MPI)
+	make_mpi_version(libquick libquick_mpi LANGUAGES Fortran INSTALL)
```

Comment:
Trivial question: If `MPI` is passed by the command line `-DMPI=TRUE`, what's the difference between `if(MPI)` and `if(${MPI})`? Seems both work?

Reply:
Yes they are essentially the same.